src/Framework/TestRunner/templates/class.tpl

```diff
@@ -69,52 +72,67 @@ function __phpunit_run_isolated_test()
 
     ErrorHandler::instance()->useDeprecationTriggers($deprecationTriggers);
 
-    $test = new {className}('{name}');
+    ini_set('xdebug.scream', '0');
```

Comment:
is this line fixing a separate problem and should be separated from this PR?

Reply:
Well problem began with PHPUnit calling before and after class methods from primary process when run in separated process. After fixing the issue I stumbled on another one where those methods are called for every test while `RunClassInSeparateProcess` is defined. So yes, this PR contains fixes to before and after class methods being falsely called while using `RunTestsInSeparateProcesses`, `RunClassInSeparateProcess` and `RunInSeparateProcess` attribute.

Reply:
This line was in the source before my redesign. Copied common functionality from previous version. 

```
$test->setData('{dataName}', unserialize('{data}'));
$test->setDependencyInput(unserialize('{dependencyInput}'));
```

These lines I don't known what they do. Are they for proxing primary process global state?
Our project now runs all the tests properly without any issues. We have suites without and with separated process. Also mixed tests.

Reply:
> ```
> $test->setData('{dataName}', unserialize('{data}'));
> $test->setDependencyInput(unserialize('{dependencyInput}'));
> ```
> 
> These lines I don't known what they do.

The pass data from data providers and depended-upon tests to the test.